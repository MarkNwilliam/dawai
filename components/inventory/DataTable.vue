<template>
  <v-card class="mt-10 pb-5" outlined flat>
    <v-sheet
      class="v-sheet--offset mx-auto px-4 py-8"
      color="primary"
      min-height="100px"
      max-width="calc(100% - 64px)"
      elevation="6"
      dark
    >
      <span class="display-2 font-weight-thin">Inventory</span>
    </v-sheet>
    <v-card-title>
      <v-text-field
        v-model="search"
        label="Search Medcine"
        type="text"
        :clear-icon="mdiClose"
        clearable
        :prepend-inner-icon="mdiMagnify"
        outlined
      ></v-text-field>
      <v-spacer></v-spacer>
      <v-btn class="mr-2" large text color="primary"
        >Add Medicine <v-icon right>{{ mdiPlusCircleOutline }}</v-icon></v-btn
      >
      <v-btn class="ml-2" large text color="primary"
        >Refresh <v-icon right>{{ mdiRefresh }}</v-icon></v-btn
      >
    </v-card-title>
    <v-data-table
      :headers="headers"
      :items="employees"
      :page.sync="page"
      :search="search"
      :items-per-page="itemsPerPage"
      hide-default-footer
      multi-sort
      @page-count="pageCount = $event"
    >
      <template v-slot:item.price="{ item }">
        <span
          >{{ item.price.toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,') }}
          <sup>SDG</sup></span
        >
      </template>
      <template v-slot:item.availability="{ item }">
        <v-chip v-if="item.availability" color="success">
          <v-icon>{{ mdiCheckCircleOutline }}</v-icon>
        </v-chip>
        <v-chip v-else class="white--text" color="danger"
          ><v-icon>{{ mdiCloseCircleOutline }}</v-icon></v-chip
        >
      </template>
      <template v-slot:item.action="{ item }">
        <v-btn text icon color="green">
          <v-icon small>
            {{ mdiPencil }}
          </v-icon>
        </v-btn>
        <v-btn text icon color="red">
          <v-icon small>
            {{ mdiDeleteForever }}
          </v-icon>
        </v-btn>
      </template>
      <template v-slot:no-data>
        <v-btn color="primary" @click="initialize">
          Reset
        </v-btn>
      </template>

      <template v-slot:no-results>
        Oops! Medicine doesn't exist!
        <v-btn class="mr-2" small text color="primary"
          >Add Medicine <v-icon right>{{ mdiPlusCircleOutline }}</v-icon></v-btn
        >
      </template>
    </v-data-table>
    <div class="text-center pt-2">
      <v-pagination
        v-model="page"
        :next-icon="mdiChevronRight"
        color="primary"
        circle
        :prev-icon="mdiChevronLeft"
        :length="pageCount"
      />
    </div>
  </v-card>
</template>

<script>
import {
  mdiChevronRight,
  mdiChevronLeft,
  mdiPencil,
  mdiDeleteForever,
  mdiPlusCircleOutline,
  mdiRefresh,
  mdiMagnify,
  mdiClose,
  mdiCheckCircleOutline,
  mdiCloseCircleOutline
} from '@mdi/js'
export default {
  data: () => ({
    mdiChevronRight,
    mdiChevronLeft,
    mdiPencil,
    mdiDeleteForever,
    mdiPlusCircleOutline,
    mdiRefresh,
    mdiMagnify,
    mdiClose,
    mdiCheckCircleOutline,
    mdiCloseCircleOutline,
    page: 1,
    pageCount: 0,
    itemsPerPage: 5,
    search: '',
    headers: [
      {
        text: 'Name',
        align: 'left',
        sortable: true,
        value: 'name'
      },
      { text: 'Quantity', align: 'center', value: 'quantity' },
      { text: 'Price', align: 'center', value: 'price' },
      { text: 'Availability', align: 'right', value: 'availability' },
      { text: 'Actions', value: 'action', sortable: false, align: 'center' }
    ],
    employees: [
      {
        name: 'Panadol',
        quantity: 15,
        price: 40,
        availability: true
      },
      {
        name: 'Ibuprofen',
        quantity: 28,
        price: 60,
        availability: true
      },
      {
        name: 'Paracetamol',
        quantity: 43,
        price: 55,
        availability: true
      },
      {
        name: 'Panadol Extra',
        quantity: 89,
        price: 800,
        availability: false
      },
      {
        name: 'Panadol Lite',
        quantity: 99,
        price: 1200,
        availability: false
      }
    ]
  })
}
</script>

<style scoped>
.v-sheet--offset {
  top: -24px;
  position: relative;
}
</style>
