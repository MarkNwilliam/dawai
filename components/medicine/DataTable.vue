<template>
  <v-card class="mt-10 pb-5 slight-br" outlined flat>
    <v-sheet
      class="text-center v-sheet--offset mx-auto py-8 heavy-br"
      color="primary"
      min-height="100px"
      max-width="calc(100% - 64px)"
      elevation="6"
      dark
    >
      <span class="display-2 font-weight-thin">Medicine</span>
    </v-sheet>
    <v-card-title>
      <v-row class="mx-5" align="center" justify="center">
        <v-text-field
          v-model="search"
          label="Search Medcine"
          type="text"
          :clear-icon="mdiClose"
          :disabled="loading"
          :prepend-inner-icon="mdiMagnify"
          clearable
          outlined
          rounded
        ></v-text-field>
        <v-spacer></v-spacer>

        <v-btn
          large
          text
          color="primary"
          :disabled="loading"
          :loading="loading"
          rounded
          @click.stop.prevent="beginLoading"
          >Refresh <v-icon right>{{ mdiRefresh }}</v-icon></v-btn
        >
        <v-dialog v-model="dialog" max-width="500px">
          <template v-slot:activator="{ on }"
            ><v-btn
              :disabled="loading"
              large
              text
              color="primary"
              rounded
              v-on="on"
              >Add Item <v-icon right>{{ mdiPlusCircleOutline }}</v-icon></v-btn
            >
          </template>
          <v-card>
            <v-card-title>
              <span class="headline">{{ formTitle }}</span>
            </v-card-title>
            <v-card-text class="pb-0">
              <v-row dense>
                <v-col cols="12">
                  <v-text-field
                    v-model="editedItem.name"
                    label="Item Name"
                    :prepend-inner-icon="mdiRenameBox"
                    type="text"
                    outlined
                  ></v-text-field>
                </v-col>
                <v-col cols="12">
                  <v-text-field
                    v-model="editedItem.price"
                    label="Price"
                    type="number"
                    :prepend-inner-icon="mdiCurrencyUsd"
                    outlined
                  ></v-text-field>
                </v-col>
                <v-col cols="12" sm="6">
                  <v-menu
                    ref="menu1"
                    v-model="menu1"
                    :close-on-content-click="false"
                    :return-value.sync="editedItem.productionDate"
                    transition="scale-transition"
                    offset-y
                    min-width="290px"
                  >
                    <template v-slot:activator="{ on }">
                      <v-text-field
                        v-model="editedItem.productionDate"
                        label="Production Date"
                        :prepend-inner-icon="mdiCalendarClock"
                        readonly
                        outlined
                        v-on="on"
                      ></v-text-field>
                    </template>
                    <v-date-picker
                      v-model="editedItem.productionDate"
                      :next-icon="mdiChevronRight"
                      :prev-icon="mdiChevronLeft"
                      no-title
                      scrollable
                    >
                      <div class="flex-grow-1"></div>
                      <v-btn text color="primary" @click="menu1 = false"
                        >Cancel</v-btn
                      >
                      <v-btn
                        text
                        color="primary"
                        @click="$refs.menu1.save(editedItem.productionDate)"
                        >OK</v-btn
                      >
                    </v-date-picker>
                  </v-menu>
                </v-col>
                <v-col cols="12" sm="6">
                  <v-menu
                    ref="menu2"
                    v-model="menu2"
                    :close-on-content-click="false"
                    :return-value.sync="editedItem.expiry"
                    transition="scale-transition"
                    offset-y
                    min-width="290px"
                  >
                    <template v-slot:activator="{ on }">
                      <v-text-field
                        v-model="editedItem.expiry"
                        label="Expiry Date"
                        :prepend-inner-icon="mdiCalendarAlert"
                        readonly
                        outlined
                        v-on="on"
                      ></v-text-field>
                    </template>
                    <v-date-picker
                      v-model="editedItem.expiry"
                      :next-icon="mdiChevronRight"
                      :prev-icon="mdiChevronLeft"
                      no-title
                      scrollable
                    >
                      <div class="flex-grow-1"></div>
                      <v-btn text color="primary" @click="menu2 = false"
                        >Cancel</v-btn
                      >
                      <v-btn
                        text
                        color="primary"
                        @click="$refs.menu2.save(editedItem.expiry)"
                        >OK</v-btn
                      >
                    </v-date-picker>
                  </v-menu>
                </v-col>
                <v-col cols="12">
                  <v-select
                    v-model="editedItem.category"
                    :items="inventoryCategories"
                    label="Category"
                    type="text"
                    :append-icon="mdiChevronDown"
                    :prepend-inner-icon="mdiShapeOutline"
                    outlined
                  ></v-select>
                </v-col>
                <v-col cols="12">
                  <v-slider
                    v-model="editedItem.quantity"
                    label="Quantity"
                    max="100"
                    min="0"
                    thumb-label
                    step="1"
                  ></v-slider>
                </v-col>
                <v-col cols="6">
                  <v-checkbox
                    v-model="editedItem.availability"
                    label="Available"
                    color="primary"
                    :on-icon="mdiCheckboxMarkedCircleOutline"
                    :off-icon="mdiCheckboxBlankCircleOutline"
                  ></v-checkbox>
                </v-col>
                <v-col cols="6">
                  <v-checkbox
                    v-model="editedItem.perscription"
                    label="Requires Perscription"
                    color="primary"
                    :on-icon="mdiCheckboxMarkedCircleOutline"
                    :off-icon="mdiCheckboxBlankCircleOutline"
                  ></v-checkbox>
                </v-col>
              </v-row>
            </v-card-text>

            <v-card-actions class="pa-5">
              <div class="flex-grow-1"></div>
              <v-btn color="red" text @click="close"
                >Cancel <v-icon right>{{ mdiClose }}</v-icon></v-btn
              >
              <v-btn color="primary" text @click="save"
                >Save <v-icon right>{{ mdiContentSave }}</v-icon></v-btn
              >
            </v-card-actions>
          </v-card>
        </v-dialog>
      </v-row>
    </v-card-title>
    <v-data-table
      :headers="headers"
      :items="inventory"
      :page.sync="page"
      :search="search"
      :loading="loading"
      :loading-text="'Refreshing your inventory...'"
      multi-sort
      hide-default-footer
      @page-count="pageCount = $event"
    >
      <template v-slot:item.perscription="{ item }">
        <v-tooltip v-if="item.perscription" bottom>
          <template v-slot:activator="{ on }">
            <v-chip color="transparent" dark v-on="on">
              <v-icon color="warning">{{ mdiAlertCircle }}</v-icon>
            </v-chip>
          </template>
          <span>Required</span>
        </v-tooltip>
        <v-tooltip v-else bottom>
          <template v-slot:activator="{ on }">
            <v-chip color="transparent" dark v-on="on">
              <v-icon color="rgb(128,128,128)">{{ mdiCloseCircle }}</v-icon>
            </v-chip>
          </template>
          <span>Not Required</span>
        </v-tooltip>
      </template>
      <template v-slot:item.expiry="{ item }">
        <v-progress-linear
          v-if="getExpiry(item.productionDate, item.expiry) >= 75"
          color="green lighten-1"
          height="10"
          rounded
          striped
          :value="getExpiry(item.productionDate, item.expiry)"
        ></v-progress-linear>
        <v-progress-linear
          v-else-if="
            getExpiry(item.productionDate, item.expiry) <= 74 &&
              getExpiry(item.productionDate, item.expiry) >= 35
          "
          color="orange darken-1"
          height="10"
          rounded
          striped
          :value="getExpiry(item.productionDate, item.expiry)"
        ></v-progress-linear>
        <v-progress-linear
          v-else
          color="red darken-1"
          height="10"
          rounded
          striped
          :value="getExpiry(item.productionDate, item.expiry)"
        ></v-progress-linear>
      </template>
      <template v-slot:item.price="{ item }">
        <span>{{ item.price }} <sup>SDG</sup></span>
      </template>
      <template v-slot:item.availability="{ item }">
        <v-tooltip v-if="item.availability" bottom>
          <template v-slot:activator="{ on }">
            <v-chip color="transparent" dark v-on="on">
              <v-icon color="success">{{ mdiCheckCircle }}</v-icon>
            </v-chip>
          </template>
          <span>Available</span>
        </v-tooltip>
        <v-tooltip v-else bottom>
          <template v-slot:activator="{ on }">
            <v-chip color="transparent" dark v-on="on">
              <v-icon color="danger">{{ mdiCloseCircle }}</v-icon>
            </v-chip>
          </template>
          <span>Unavailable</span>
        </v-tooltip>
      </template>
      <template v-slot:item.action="{ item }">
        <v-btn
          text
          icon
          :disabled="loading"
          color="green"
          @click="editItem(item)"
        >
          <v-icon small>
            {{ mdiPencil }}
          </v-icon>
        </v-btn>
        <v-btn
          text
          icon
          :disabled="loading"
          color="red"
          @click="deleteItem(item)"
        >
          <v-icon small>
            {{ mdiDeleteForever }}
          </v-icon>
        </v-btn>
      </template>
      <template v-slot:no-data>
        Nothing in Inventory!
      </template>

      <template v-slot:no-results>
        Oops! Item doesn't exist!
      </template>
    </v-data-table>
    <div class="text-center pt-2">
      <v-pagination
        v-model="page"
        :next-icon="mdiChevronRight"
        color="primary"
        circle
        :prev-icon="mdiChevronLeft"
        :length="pageCount"
      />
    </div>
  </v-card>
</template>

<script>
import {
  mdiChevronRight,
  mdiChevronLeft,
  mdiPencil,
  mdiDeleteForever,
  mdiPlusCircleOutline,
  mdiChevronDown,
  mdiRefresh,
  mdiMagnify,
  mdiClose,
  mdiCheckCircle,
  mdiCloseCircle,
  mdiCheckboxMarkedCircleOutline,
  mdiCheckboxBlankCircleOutline,
  mdiContentSave,
  mdiCurrencyUsd,
  mdiCalendarAlert,
  mdiRenameBox,
  mdiShapeOutline,
  mdiCalendarClock,
  mdiAlertCircle
} from '@mdi/js'
export default {
  data: () => ({
    dialog: false,
    editedIndex: -1,
    editedItem: {
      name: '',
      category: '',
      perscription: false,
      quantity: 0,
      price: 0,
      productionDate: '',
      expiry: '',
      availability: false
    },
    defaultItem: {
      name: '',
      category: '',
      perscription: false,
      quantity: 0,
      price: 0,
      productionDate: '',
      expiry: '',
      availability: false
    },
    mdiChevronRight,
    mdiChevronLeft,
    mdiPencil,
    mdiDeleteForever,
    mdiPlusCircleOutline,
    mdiChevronDown,
    mdiRefresh,
    mdiMagnify,
    mdiClose,
    mdiCheckCircle,
    mdiCloseCircle,
    mdiCheckboxMarkedCircleOutline,
    mdiCheckboxBlankCircleOutline,
    mdiContentSave,
    mdiCurrencyUsd,
    mdiCalendarAlert,
    mdiRenameBox,
    mdiShapeOutline,
    mdiCalendarClock,
    mdiAlertCircle,
    menu1: false,
    menu2: false,
    page: 1,
    pageCount: 0,
    search: '',
    loading: false,
    inventoryCategories: [
      'Pill',
      'Powder',
      'Cream',
      'Natural',
      'Paste',
      'Injection',
      'Inhalative',
      'Drop',
      'Suppository',
      'Other'
    ],
    headers: [
      {
        text: 'Name',
        align: 'left',
        value: 'name'
      },
      { text: 'Category', align: 'center', value: 'category' },
      {
        text: 'Requires Perscription',
        align: 'center',
        value: 'perscription'
      },
      { text: 'Quantity', align: 'center', value: 'quantity' },
      { text: 'Price', align: 'center', value: 'price' },
      { text: 'Expiry', align: 'center', value: 'expiry' },
      { text: 'Availability', align: 'right', value: 'availability' },
      { text: 'Actions', value: 'action', sortable: false, align: 'center' }
    ],
    inventory: [
      {
        name: 'Panadol',
        category: 'Pill',
        perscription: false,
        quantity: 15,
        price: 40,
        productionDate: '2018-01-01',
        expiry: '2020-01-01',
        availability: true
      },
      {
        name: 'Ibuprofen',
        category: 'Pill',
        perscription: false,
        quantity: 28,
        price: 60,
        productionDate: '2018-01-01',
        expiry: '2020-01-01',
        availability: true
      },
      {
        name: 'Paracetamol',
        category: 'Pill',
        perscription: false,
        quantity: 43,
        price: 55,
        productionDate: '2019-01-01',
        expiry: '2020-01-01',
        availability: true
      },
      {
        name: 'Panadol Extra',
        category: 'Pill',
        perscription: true,
        quantity: 89,
        price: 800,
        productionDate: '2019-09-01',
        expiry: '2020-01-01',
        availability: false
      },
      {
        name: 'Panadol Lite',
        category: 'Pill',
        perscription: true,
        quantity: 99,
        price: 1200,
        productionDate: '2019-06-01',
        expiry: '2020-01-01',
        availability: false
      }
    ]
  }),
  computed: {
    formTitle() {
      return this.editedIndex === -1 ? 'New Item Entry' : 'Edit Item'
    }
  },
  methods: {
    beginLoading() {
      this.loading = true
      setTimeout(() => {
        this.loading = false
      }, 5000)
    },

    editItem(item) {
      this.editedIndex = this.inventory.indexOf(item)
      this.editedItem = Object.assign({}, item)
      this.dialog = true
    },

    deleteItem(item) {
      const index = this.inventory.indexOf(item)
      confirm('Are you sure you want to delete this item?') &&
        this.inventory.splice(index, 1)
    },

    close() {
      this.dialog = false
      setTimeout(() => {
        this.editedItem = Object.assign({}, this.defaultItem)
        this.editedIndex = -1
      }, 300)
    },

    save() {
      if (this.editedIndex > -1) {
        Object.assign(this.inventory[this.editedIndex], this.editedItem)
      } else {
        this.inventory.push(this.editedItem)
      }
      this.close()
    },

    getExpiry(productionDate, expiryDate) {
      const oneDay = 1000 * 60 * 60 * 24
      const duration = Math.abs(
        Math.round((new Date(expiryDate) - new Date(productionDate)) / oneDay)
      )
      const now = Math.abs(
        Math.round((new Date(productionDate) - new Date()) / oneDay)
      )
      return 100 - Math.round((now / duration) * 100)
    }
  }
}
</script>

<style scoped>
.v-sheet--offset {
  top: -24px;
  position: relative;
}
</style>
