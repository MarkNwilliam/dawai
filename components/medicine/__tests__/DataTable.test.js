// Libraries
import Vuetify from 'vuetify'

// Component
import { mount, createLocalVue } from '@vue/test-utils'
import DataTable from '../DataTable'

// Utilities
const localVue = createLocalVue()

describe('DataTable.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should render data table with static data', (done) => {
    const wrapper = mount(DataTable, {
      localVue,
      vuetify
    })

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()

    done()
  })
})
