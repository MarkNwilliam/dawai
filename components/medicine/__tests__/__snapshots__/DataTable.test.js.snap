// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.vue should render data table with static data 1`] = `"<v-card outlined=\\"\\" flat=\\"\\" class=\\"mt-10 pb-5 slight-br\\"><v-sheet color=\\"primary\\" min-height=\\"100px\\" max-width=\\"calc(100% - 64px)\\" elevation=\\"6\\" dark=\\"\\" class=\\"text-center v-sheet--offset mx-auto py-8 heavy-br\\"><span class=\\"display-2 font-weight-thin\\">Medicine</span></v-sheet> <v-card-title><v-row align=\\"center\\" justify=\\"center\\" class=\\"mx-5\\"><v-text-field label=\\"Search Medcine\\" type=\\"text\\" clear-icon=\\"M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z\\" prepend-inner-icon=\\"M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z\\" clearable=\\"\\" outlined=\\"\\" rounded=\\"\\"></v-text-field> <v-spacer></v-spacer> <v-btn large=\\"\\" text=\\"\\" color=\\"primary\\" rounded=\\"\\">Refresh <v-icon right=\\"\\">M17.65,6.35C16.2,4.9 14.21,4 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20C15.73,20 18.84,17.45 19.73,14H17.65C16.83,16.33 14.61,18 12,18A6,6 0 0,1 6,12A6,6 0 0,1 12,6C13.66,6 15.14,6.69 16.22,7.78L13,11H20V4L17.65,6.35Z</v-icon></v-btn> <v-dialog max-width=\\"500px\\"> <v-card><v-card-title><span class=\\"headline\\">New Item Entry</span></v-card-title> <v-card-text class=\\"pb-0\\"><v-row dense=\\"\\"><v-col cols=\\"12\\"><v-text-field label=\\"Item Name\\" prepend-inner-icon=\\"M18,17H10.5L12.5,15H18M6,17V14.5L13.88,6.65C14.07,6.45 14.39,6.45 14.59,6.65L16.35,8.41C16.55,8.61 16.55,8.92 16.35,9.12L8.47,17M19,3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3Z\\" type=\\"text\\" outlined=\\"\\"></v-text-field></v-col> <v-col cols=\\"12\\"><v-text-field label=\\"Price\\" type=\\"number\\" prepend-inner-icon=\\"M7,15H9C9,16.08 10.37,17 12,17C13.63,17 15,16.08 15,15C15,13.9 13.96,13.5 11.76,12.97C9.64,12.44 7,11.78 7,9C7,7.21 8.47,5.69 10.5,5.18V3H13.5V5.18C15.53,5.69 17,7.21 17,9H15C15,7.92 13.63,7 12,7C10.37,7 9,7.92 9,9C9,10.1 10.04,10.5 12.24,11.03C14.36,11.56 17,12.22 17,15C17,16.79 15.53,18.31 13.5,18.82V21H10.5V18.82C8.47,18.31 7,16.79 7,15Z\\" outlined=\\"\\"></v-text-field></v-col> <v-col cols=\\"12\\" sm=\\"6\\"><v-menu return-value=\\"\\" transition=\\"scale-transition\\" offset-y=\\"\\" min-width=\\"290px\\"> <v-date-picker next-icon=\\"M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z\\" prev-icon=\\"M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z\\" no-title=\\"\\" scrollable=\\"\\"><div class=\\"flex-grow-1\\"></div> <v-btn text=\\"\\" color=\\"primary\\">Cancel</v-btn> <v-btn text=\\"\\" color=\\"primary\\">OK</v-btn></v-date-picker></v-menu></v-col> <v-col cols=\\"12\\" sm=\\"6\\"><v-menu return-value=\\"\\" transition=\\"scale-transition\\" offset-y=\\"\\" min-width=\\"290px\\"> <v-date-picker next-icon=\\"M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z\\" prev-icon=\\"M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z\\" no-title=\\"\\" scrollable=\\"\\"><div class=\\"flex-grow-1\\"></div> <v-btn text=\\"\\" color=\\"primary\\">Cancel</v-btn> <v-btn text=\\"\\" color=\\"primary\\">OK</v-btn></v-date-picker></v-menu></v-col> <v-col cols=\\"12\\"><v-select items=\\"Pill,Powder,Cream,Natural,Paste,Injection,Inhalative,Drop,Suppository,Other\\" label=\\"Category\\" type=\\"text\\" append-icon=\\"M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z\\" prepend-inner-icon=\\"M11,13.5V21.5H3V13.5H11M9,15.5H5V19.5H9V15.5M12,2L17.5,11H6.5L12,2M12,5.86L10.08,9H13.92L12,5.86M17.5,13C20,13 22,15 22,17.5C22,20 20,22 17.5,22C15,22 13,20 13,17.5C13,15 15,13 17.5,13M17.5,15A2.5,2.5 0 0,0 15,17.5A2.5,2.5 0 0,0 17.5,20A2.5,2.5 0 0,0 20,17.5A2.5,2.5 0 0,0 17.5,15Z\\" outlined=\\"\\"></v-select></v-col> <v-col cols=\\"12\\"><v-slider label=\\"Quantity\\" max=\\"100\\" min=\\"0\\" thumb-label=\\"\\" step=\\"1\\"></v-slider></v-col> <v-col cols=\\"6\\"><v-checkbox label=\\"Available\\" color=\\"primary\\" on-icon=\\"M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2,4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z\\" off-icon=\\"M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z\\"></v-checkbox></v-col> <v-col cols=\\"6\\"><v-checkbox label=\\"Requires Perscription\\" color=\\"primary\\" on-icon=\\"M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2,4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z\\" off-icon=\\"M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z\\"></v-checkbox></v-col></v-row></v-card-text> <v-card-actions class=\\"pa-5\\"><div class=\\"flex-grow-1\\"></div> <v-btn color=\\"red\\" text=\\"\\">Cancel <v-icon right=\\"\\">M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z</v-icon></v-btn> <v-btn color=\\"primary\\" text=\\"\\">Save <v-icon right=\\"\\">M15,9H5V5H15M12,19A3,3 0 0,1 9,16A3,3 0 0,1 12,13A3,3 0 0,1 15,16A3,3 0 0,1 12,19M17,3H5C3.89,3 3,3.9 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V7L17,3Z</v-icon></v-btn></v-card-actions></v-card></v-dialog></v-row></v-card-title> <v-data-table headers=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object]\\" page=\\"1\\" search=\\"\\" loading-text=\\"Refreshing your inventory...\\" multi-sort=\\"\\" hide-default-footer=\\"\\"></v-data-table> <div class=\\"text-center pt-2\\"><v-pagination next-icon=\\"M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z\\" color=\\"primary\\" circle=\\"\\" prev-icon=\\"M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z\\" length=\\"0\\"></v-pagination></div></v-card>"`;
